make_prg/from_msa/cluster_sequences.py

```diff
@@ -16,32 +16,6 @@
 ClusteredSeqs = List[Sequences]
 KmerIDs = Dict[Sequence, int]
 
-
-class ClusteringResult(object):
-    """
-    Stores the result of clustering as a ClusteredIDs object.
-    If clustering was not meaningful (e.g. one sequence per cluster),
-    returns a set of sequences instead that can be directly used as set of alternative
-    alleles of the variant site under construction.
-    """
-
-    def __init__(
-        self,
-        clustered_ids: Optional[ClusteredIDs],
-        sequences: Optional[Sequences],
-    ):
-        mutually_exclusive = (clustered_ids is None) ^ (sequences is None)
-        assert (
-            mutually_exclusive
-        ), "Input arguments must be mutually exclusively set to 'None'"
-        self.clustered_ids = clustered_ids
-        self.sequences = sequences
-
-    @property
-    def no_clustering(self):
-        return self.clustered_ids is None
```

Comment:
When I wrote it I found this class a little clumsy, but what is your reasoning for removing it? How do you now test if clustering produced a meaningful clustering?

Reply:
The behaviour of this class is different from the `skip_clustering` method in `PrgBuilder` (in `prg_builder.py`). This code is where we've decided not to skip clustering, we do kmeans and for some reason or other (mainly: too few long-enough sequences and as many clusters as sequences) there is no meaningful clustering. In that case we return the sequences directly and they get enumerated as alleles directly. As long as that behaviour is maintained somehow I would OK this change

Reply:
You are right, this class was added back in the last commits, with some changes though. In a part of the new code, we always need the clustering, so the mutual exclusivity of the attributes was removed, and we always know the `clustered_ids` of a `ClusteringResult`. Optionally, if the clustering was not significant, then `sequences` won't be `None`, and we can get the precomputed set of alleles directly. I've added some other trivial methods required by testings and other code. Please recheck the new version

Reply:
You will have to see the `Files changed` section in this PR to see this class back